silkworm/sentry/common/timeout.hpp

```diff
@@ -25,28 +25,15 @@
 
 namespace silkworm::sentry::common {
 
-class Timeout {
-  public:
-    explicit Timeout(std::chrono::milliseconds duration) : duration_(duration) {}
+struct Timeout {
+    Timeout() = delete;
 
-    Timeout(const Timeout&) = delete;
-    Timeout& operator=(const Timeout&) = delete;
-
-    [[nodiscard]] boost::asio::awaitable<void> schedule() const;
-    boost::asio::awaitable<void> operator()() const { return schedule(); }
-
-    static boost::asio::awaitable<void> after(std::chrono::milliseconds duration) {
-        Timeout timeout(duration);
-        co_await timeout.schedule();
-    }
+    static boost::asio::awaitable<void> after(std::chrono::milliseconds duration);
 
     class ExpiredError : public std::runtime_error {
       public:
         ExpiredError() : std::runtime_error("Timeout has expired") {}
     };
-
-  private:
-    std::chrono::milliseconds duration_;
```

Comment:
In order to simplify this even further, why not having just one plain function:
```
boost::asio::awaitable<void> timeout(std::chrono::milliseconds duration);
```
and maybe renaming `ExpiredError` as `TimeoutExpiredError`.